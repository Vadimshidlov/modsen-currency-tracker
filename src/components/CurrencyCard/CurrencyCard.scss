@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.currency-card__containter {
    display: flex;
    column-gap: 30px;
    align-items: center;
    width: 520px;
    background-color: colors.$currency_card_background;
    padding: 36px 32px;
    border-radius: 8px;
    border: 1px solid colors.$currency_card_border;
    transition: all 1s;

    &:hover {
        opacity: 0.7;
        cursor: pointer;
    }

    .currency-card__info {
        display: flex;
        flex-direction: column;
        align-items: start;
        row-gap: 10px;

        .currency-card__titile {
            @include fonts.PoppinsRegularFont;
            color: colors.$color_grey;
            font-size: fonts.$font-size_35;
            line-height: 49.37px;
        }
        .currency-card__percent {
            @include fonts.PoppinsLight;
            color: colors.$color_grey_medium-opacity;
            font-size: fonts.$font-size_32;
            line-height: 41.14px;
        }
    }
}
