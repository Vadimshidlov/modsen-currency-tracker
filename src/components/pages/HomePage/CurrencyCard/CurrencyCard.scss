@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;


.currency-card__container {
  display: flex;
  column-gap: 30px;
  align-items: center;
  width: 520px;
  background-color: colors.$currency_card_background;
  padding: 36px 32px;
  border-radius: 8px;
  border: 1px solid colors.$currency_card_border;
  transition: all 1s;
  outline: none;

  @media (max-width: 1120px) {
    width: 100%;
    padding: 20px 16px;
  }

  @media (max-width: 450px) {
    column-gap: 12px;
    padding: 8px 12px;
  }

  &:hover {
    opacity: 0.7;
    cursor: pointer;
  }

  .currency-card__icon {
    width: 80px;
    height: 80px;

    @media (max-width: 1120px) {
      width: 60px;
      height: 60px;
    }

    @media (max-width: 800px) {
      width: 45px;
      height: 45px;
    }

    @media (max-width: 500px) {
      width: 30px;
      height: 30px;
    }
  }

  .currency-card__info {
    display: flex;
    flex-direction: column;
    align-items: start;
    //row-gap: 10px;

    .currency-card__titile {
      @include fonts.PoppinsRegularFont;
      color: colors.$color_grey;
      font-size: fonts.$font-size_35;
      line-height: 49.37px;

      @media (max-width: 1120px) {
        font-size: fonts.$font-size_25;
        line-height: 35px;
      }

      @media (max-width: 900px) {
        font-size: fonts.$font-size_20;
        line-height: 30px;
      }

      @media (max-width: 500px) {
        font-size: fonts.$font-size_13;
        line-height: 18px;
      }
    }

    .currency-card__percent {
      @include fonts.PoppinsLight;
      color: colors.$color_grey_medium-opacity;
      font-size: fonts.$font-size_32;
      line-height: 41.14px;

      @media (max-width: 1120px) {
        font-size: fonts.$font-size_25;
        line-height: 35px;
      }

      @media (max-width: 900px) {
        font-size: fonts.$font-size_20;
        line-height: 30px;
      }

      @media (max-width: 500px) {
        font-size: fonts.$font-size_12;
        line-height: 15px;
      }

      /* @media (max-width: 420px) {
         font-size: fonts.$font-size_12;
         line-height: 15px;
       }*/
    }
  }
}
