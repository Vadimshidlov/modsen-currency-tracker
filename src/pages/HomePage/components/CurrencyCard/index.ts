export { default } from "@/pages/HomePage/components/CurrencyCard/CurrencyCard";
