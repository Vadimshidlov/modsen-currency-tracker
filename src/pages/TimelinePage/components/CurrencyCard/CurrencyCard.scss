@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;

.chart-card__container {
    align-self: start;
    display: flex;
    column-gap: 30px;
    align-items: center;
    width: 520px;
    padding: 36px 32px;
    border-radius: 8px;
    background: transparent;
    transition: all 1s;
    outline: none;
    margin-bottom: 20px;

    @media (max-width: 1120px) {
        width: 100%;
        padding: 20px 16px;
    }

    @media (max-width: 450px) {
        column-gap: 12px;
        padding: 8px 12px;
        margin-bottom: 10px;
    }

    &:hover {
        opacity: 0.7;
        cursor: pointer;
    }

    .chart-card__icon {
        width: 80px;
        height: 80px;

        @media (max-width: 1120px) {
            width: 60px;
            height: 60px;
        }

        @media (max-width: 800px) {
            width: 45px;
            height: 45px;
        }

        @media (max-width: 500px) {
            width: 30px;
            height: 30px;
        }
    }

    .chart-card__info {
        display: flex;
        flex-direction: column;
        align-items: start;

        .chart-card__titile {
            @include fonts.PoppinsRegularFont;
            color: colors.$color_grey;
            font-size: fonts.$font-size_35;
            line-height: 49.37px;

            @media (max-width: 1120px) {
                font-size: fonts.$font-size_25;
                line-height: 35px;
            }

            @media (max-width: 900px) {
                font-size: fonts.$font-size_20;
                line-height: 30px;
            }

            @media (max-width: 500px) {
                font-size: fonts.$font-size_13;
                line-height: 18px;
            }
        }

        .chart-card__percent {
            @include fonts.PoppinsLight;
            color: colors.$color_grey_medium-opacity;
            font-size: fonts.$font-size_16;
            line-height: 41.14px;

            @media (max-width: 1120px) {
                font-size: fonts.$font-size_25;
                line-height: 35px;
            }

            @media (max-width: 900px) {
                font-size: fonts.$font-size_20;
                line-height: 30px;
            }

            @media (max-width: 500px) {
                font-size: fonts.$font-size_12;
                line-height: 15px;
            }
        }
    }

    &.light {
        .chart-card__info {
            color: colors.$black_background;

            .chart-card__titile {
                color: colors.$black_background;
            }

            .chart-card__percent {
                color: colors.$black_background;
            }
        }
    }
}
