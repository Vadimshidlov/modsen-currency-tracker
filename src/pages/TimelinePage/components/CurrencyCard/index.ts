export { default } from "@/pages/TimelinePage/components/CurrencyCard/CurrencyCard";
