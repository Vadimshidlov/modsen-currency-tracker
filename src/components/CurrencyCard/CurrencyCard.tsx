import React from "react";
import Text from "@/components/Text/Text";
import "@/components/CurrencyCard/CurrencyCard.scss";
import ArgentinePesoIcon from "@/assets/svg/currency/argentine_peso.svg";
import AustralianDollarIcon from "@/assets/svg/currency/australian-dollar.svg";
import BitcoinIcon from "@/assets/svg/currency/bitcoin.svg";
import CanadianDollarIcon from "@/assets/svg/currency/canadian-dollar.svg";
import CommercialDollarIcon from "@/assets/svg/currency/Rectangle 3.svg";
import EuroIcon from "@/assets/svg/currency/euro.svg";
import YenIcon from "@/assets/svg/currency/yen.svg";
import YuanIcon from "@/assets/svg/currency/yuan.svg";
import { useActions } from "@/store/hooks/useActions";
import { useTypedSelectorHook } from "@/store/hooks/useTypedSelector";

export type CurrencyPropsType = {
    currencyCode: string;
    value: number;
};

export default function CurrencyCard({ currencyCode, value }: CurrencyPropsType) {
    let CurrencyIcon;
    let currencyTitle = "";
    const currencyValue = +value.toFixed(2);
    // const currencyValue = value;

    const { currency } = useTypedSelectorHook((state) => state.currency);

    const { openModalWindow } = useActions();

    const openModalHandler = () => {
        if (currencyTitle) {
            openModalWindow(currency, currencyTitle, value, currencyCode);
        }
    };

    switch (currencyCode) {
        case "USD":
            CurrencyIcon = CommercialDollarIcon;
            currencyTitle = "Commercial Dollar";
            break;
        case "CAD":
            CurrencyIcon = CanadianDollarIcon;
            currencyTitle = "Canadian Dollar";

            break;
        case "AUD":
            CurrencyIcon = AustralianDollarIcon;
            currencyTitle = "Australian Dollar";

            break;
        case "EUR":
            CurrencyIcon = EuroIcon;
            currencyTitle = "Euro";
            break;
        case "ARS":
            CurrencyIcon = ArgentinePesoIcon;
            currencyTitle = "Argentine Peso";

            break;
        case "JPY":
            CurrencyIcon = YenIcon;
            currencyTitle = "Yen";

            break;
        case "CNY":
            CurrencyIcon = YuanIcon;
            currencyTitle = "Yuan";

            break;
        case "BTC":
            CurrencyIcon = BitcoinIcon;
            currencyTitle = "Bitcoin";

            break;

        default:
            CurrencyIcon = CommercialDollarIcon;
            currencyTitle = "";
    }

    return (
        <button className="currency-card__containter" onClick={() => openModalHandler()}>
            <CurrencyIcon width={80} height={80} />
            <div className="currency-card__info">
                <Text className="currency-card__titile">{currencyTitle}</Text>
                <Text className="currency-card__percent">{`R$ ${currencyValue}`}</Text>
            </div>
        </button>
    );
}
