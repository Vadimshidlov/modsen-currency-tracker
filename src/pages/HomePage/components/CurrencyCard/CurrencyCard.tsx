import React from "react";
import Text from "@/pages/HomePage/components/Text/index";
import "@/pages/HomePage/components/CurrencyCard/CurrencyCard.scss";
import { useActions, useTypedSelectorHook } from "@/store/hooks/index";
import { getCurrencyIcon } from "@/utils/currency";
import { CurrencyCardPropsType } from "@/types/HomePageTypes";
import { RootStateType } from "@/store/reducers";

const selectCurrency = (state: RootStateType) => state.currency;

export default function CurrencyCard({ currencyCode, value }: CurrencyCardPropsType) {
    const { CurrencyIcon, currencyTitle } = getCurrencyIcon(currencyCode);

    const currencyValue = currencyCode === "BTC" ? value : +value.toFixed(2);

    const { currency } = useTypedSelectorHook(selectCurrency);

    const { openModalWindow } = useActions();

    const handleModalOpen = () => {
        if (currencyTitle) {
            openModalWindow(currency, currencyTitle, value, currencyCode);
        }
    };

    return (
        <button className="currency-card__container" onClick={handleModalOpen}>
            <CurrencyIcon className="currency-card__icon" />
            <div className="currency-card__info">
                <Text className="currency-card__titile">{currencyTitle}</Text>
                <Text className="currency-card__percent">{`R$ ${currencyValue}`}</Text>
            </div>
        </button>
    );
}
