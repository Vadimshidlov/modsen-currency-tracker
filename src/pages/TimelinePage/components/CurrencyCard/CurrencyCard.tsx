import React, { Component } from "react";
import Text from "@/components/Text/index";
import "@/pages/TimelinePage/components/CurrencyCard/CurrencyCard.scss";
import { CurrencyCardPropsType } from "@/types/TimeLinePageTypes";
import { getCurrencyIcon } from "@/utils/currency";

export default class CurrencyCard extends Component<CurrencyCardPropsType> {
    render() {
        const { currencyCode, theme } = this.props;
        const { CurrencyIcon, currencyTitle } = getCurrencyIcon(currencyCode);

        return (
            <div
                className={
                    theme === "Light" ? "chart-card__container light" : "chart-card__container"
                }
            >
                <CurrencyIcon className="chart-card__icon" />
                <div className="chart-card__info">
                    <Text className="chart-card__titile">{currencyTitle}</Text>
                    <Text className="chart-card__percent">{currencyCode}</Text>
                </div>
            </div>
        );
    }
}
