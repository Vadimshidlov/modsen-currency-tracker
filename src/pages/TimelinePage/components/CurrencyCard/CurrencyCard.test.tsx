import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "@/store";
import { renderWithProviders, screen } from "@/utils/rtl-utils";
import CurrencyCard from "@/pages/TimelinePage/components/CurrencyCard/index";

describe("CurrencyCard component tests", () => {
    it("should correct render component", async () => {
        renderWithProviders(<CurrencyCard currencyCode="USD" theme="Dark" />);

        expect(screen.getByText("Commercial Dollar")).toBeInTheDocument();
        expect(screen.getByText("USD")).toBeInTheDocument();
    });

    it("snapshot matches", async () => {
        const { asFragment } = renderWithProviders(
            <CurrencyCard currencyCode="USD" theme="Dark" />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("should match snapshot", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <CurrencyCard currencyCode="USD" theme="Dark" />
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
