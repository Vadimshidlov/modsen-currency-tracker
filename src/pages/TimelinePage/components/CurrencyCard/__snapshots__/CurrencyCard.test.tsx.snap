// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyCard component tests should match snapshot 1`] = `
<div
  className="chart-card__container"
>
  <svg
    className="chart-card__icon"
  />
  <div
    className="chart-card__info"
  >
    <p
      className="chart-card__titile"
    >
      Commercial Dollar
    </p>
    <p
      className="chart-card__percent"
    >
      USD
    </p>
  </div>
</div>
`;

exports[`CurrencyCard component tests snapshot matches 1`] = `
<DocumentFragment>
  <div
    class="chart-card__container"
  >
    <svg
      class="chart-card__icon"
    />
    <div
      class="chart-card__info"
    >
      <p
        class="chart-card__titile"
      >
        Commercial Dollar
      </p>
      <p
        class="chart-card__percent"
      >
        USD
      </p>
    </div>
  </div>
</DocumentFragment>
`;
